datasets/switchboard.py

```diff
@@ -0,0 +1,293 @@
+            else:
+                # ref: https://github.com/espnet/espnet/blob/master/egs/swbd/asr1/local/swbd1_map_words.pl
+
+                mapped_token = re.sub(
+                    "(|\-)^\[laughter-(.+)\](|\-)$", "\g<1>\g<2>\g<3>", token
+                )  # e.g. [laughter-story] -> story;
+                # 1 and 3 relate to preserving trailing "-"
+                mapped_token = re.sub(
+                    "^\[(.+)/.+\](|\-)$", "\g<1>\g<2>", mapped_token
+                )  # e.g. [it'n/isn't] -> it'n ... note
+                # 1st part may include partial-word stuff, which we process further below,
+                # e.g. [LEM[GUINI]-/LINGUINI]
+                # the (|\_) at the end is to accept and preserve trailing -'s.
+                mapped_token = re.sub(
+                    "^(|\-)\[[^][]+\](.+)$", "-\g<2>", mapped_token
+                )  # e.g. -[an]y , note \047 is quote;
+                # let the leading - be optional on input, as sometimes omitted.
+                mapped_token = re.sub(
+                    "^(.+)\[[^][]+\](|\-)$", "\g<1>-", mapped_token
+                )  # e.g. ab[solute]- -> ab-;
+                # let the trailing - be optional on input, as sometimes omitted.
+                mapped_token = re.sub(
+                    "([^][]+)\[.+\]$", "\g<1>", mapped_token
+                )  # e.g. ex[specially]-/especially] -> ex-
+                # which is a  mistake in the input.
+                mapped_token = re.sub(
+                    "^\{(.+)\}$", "\g<1>", mapped_token
+                )  # e.g. {yuppiedom} -> yuppiedom
+                mapped_token = re.sub(
+                    "([a-z])\[([^][])+\]([a-z])", "\g<1>-\g<3>", mapped_token
+                )  # e.g. ammu[n]it- -> ammu-it-
+                mapped_token = re.sub("_\d$", "", mapped_token)  # e.g. them_1 -> them
+
+                filtered_orth.append(mapped_token)
+
+        # do not add empty transcription segments
+        all_special = True
+        for token in filtered_orth:
+            if token.lower() not in special_tokens:
+                all_special = False
+                break
+        if all_special:
+            return ""
+
+        out = " ".join(filtered_orth)
+
+        # replace &
+        out = out.replace("AT&T's", "AT and T")
+        out = out.replace("&", " and ")
+
+        return out
+
+    def _get_rec_to_segs_map(self):
+        """
+        Returns recording to list of segments mapping
+        """
+        rec_to_segs = {}  # typing: dict[str, List[str]]
+        for trans_file in glob.glob(
+            os.path.join(self.trans_dir.get_path(), "*/*/*-trans.text")
+        ):
+            with uopen(trans_file, "rt") as f:
+                for line in f:
+                    seg_info = line.strip().split(" ", 3)  # name start end orth
+                    assert len(seg_info) == 4
+                    rec_name = seg_info[0].split("-")[0]
+                    if rec_name not in rec_to_segs:
```

Comment:
Here it's usually nicer to use a defaultdict instead of manually checking for the existence of some key, See https://docs.python.org/3/library/collections.html#collections.defaultdict